src/voice-broadcast/audio/VoiceBroadcastRecording.ts

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { VoiceRecording } from "../../audio/VoiceRecording";
+import SdkConfig from "../../SdkConfig";
+import { concat } from "../../utils/arrays";
+import { IDestroyable } from "../../utils/IDestroyable";
+
+export enum VoiceBroadcastRecordingEvents {
+    ChunkRecorded = "chunk_recorded",
+}
+
+interface ChunkRecordedPayload {
+    buffer: Uint8Array;
+    length: number;
+    size: number;
+}
+
+/**
+ * This class provides the function to seamlessly record fixed length chunks.
+ * Subscribe with on(VoiceBroadcastRecordingEvents.ChunkRecorded, (payload: ChunkRecordedPayload) => {})
+ * to retrieve chunks while recording.
+ */
+export class VoiceBroadcastRecording implements IDestroyable {
```

Comment:
This doesn't really appear to be used outside of tests?

Reply:
It will be with one of the next PRs. Trying to split up things for review…

Reply:
Then let's keep this open until the other part is ready.